osfoffline/application/background.py

```diff
@@ -47,6 +47,3 @@ def stop(self):
         RemoteSyncWorker().stop()
         LocalSyncWorker().stop()
 
-        OperationWorker().join()
```

Comment:
Now the stop function of each of the worker ensure the worker terminates so there is no need to block those methods as it already terminates before the join method is called.